### PR TITLE
mindforger: new port

### PR DESCRIPTION
**Pull request by herbygillot**
#### Description

New port for the **[MindForger](https://www.mindforger.com)** GPLv2 personal knowledge base.

###### Tested on

macOS 10.15.7 19H114
Xcode 12.3 12C33

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -d install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: editors/mindforger/Portfile [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+categories              editors office
+license                 GPL-2
+platforms               darwin macosx
+
+maintainers             {gmail.com:herby.gillot @herbygillot} \
+                        openmaintainer
+
+fetch.type              git
+
+depends_build-append    port:ccache \
```
Comment (kencu) [inline editors/mindforger/Portfile:29]: you don't require ccache, do you?

File: editors/mindforger/Portfile [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                        port:cmake
```
Comment (kencu) [inline editors/mindforger/Portfile:30]: 99.999% of the time, you want the cmake portgroup instead of this...

File: editors/mindforger/Portfile [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+configure.cmd           ${qt_qmake_cmd}
+configure.pre_args
+configure.args          -r ./mindforger.pro
+configure.env           CCACHE_DIR=${_cc_tmp_path}
+
+build.env-append        CCACHE_DIR=${_cc_tmp_path}
+
+post-extract {
+    system -W ${worksrcpath} "git submodule update --init --recursive"
```
Comment (kencu) [inline editors/mindforger/Portfile:48]: this is possible, but will not be overly popular for all the usual reasons. No option?

**Comment by kencu** [comment]
You're doing a fair amount of manual work here... macports' usual mechanisms not work for you? Looks like a 4-liner port...

**Comment by kencu** [comment]
I think I see why you're running into so much trouble. This is a qt5 project that uses qmake.

You want to be using the qmake5 PortGroup here instead of most of this manual stuff.

The qmake5 PG is not the same as the qt5 PG (the qmake5 PG includes the qt5 PG automatically).

The qmake5 PG is more like the cmake Portgroup -- it sets everything up for a qt5 build using *.pro files, and then you should find this much much simpler than what has gone on here.

**Comment by herbygillot** [comment]
OK thanks, didn't know about the `qmake5` portgroup, will take a look later.

**Comment by kencu** [comment]
After trying it, I see why you can't just easily use the qmake5 PG... there is a cmake project embedded in the deps and it turns out to be not so simple as you might hope to build a cmake subproject with qmake.

**Comment by herbygillot** [comment]
> After trying it, I see why you can't just easily use the qmake5 PG... there is a cmake project embedded in the deps and it turns out to be not so simple as you might hope to build a cmake subproject with qmake.

Yeah, there was a lot going on in that project.

**Comment by kencu** [comment]
@herbygillot -- if you pull out the unnecessary ccache forcing, probably someone could see their way clear to committing this.

It's a messy port -- not 100% it deserves to be a port, to be honest -- and if it is committed you can expect lots of complaints about this one from above.

**Comment by herbygillot** [comment]
OK @kencu, made a number of changes, primarily removing `ccache` and using the `cmake` portgroup. We'll see if it builds.

**Comment by kencu** [comment]
yeah, if this version works it will be a lot more acceptable. Thanks.

**Pull request merged**